pilot/pkg/autoregistration/internal/health/util.go

```diff
@@ -0,0 +1,50 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health
+
+import (
+	"istio.io/api/meta/v1alpha1"
+	"istio.io/istio/pilot/pkg/model/status"
+	"istio.io/istio/pkg/config"
+)
+
+// IsElegibleForHealthStatusUpdates returns true if a given WorkloadEntry
+// is allowed to receive health status updates sent by an Istio Proxy.
+//
+// Consider a workload elegible for health status updates as long as the
+// WorkloadEntryHealthCheckAnnotation is present (no matter what the value is).
+// In case the annotation is present but the value is not "true", the proxy should be allowed
+// to send health status updates, config health condition should be updated accordingly,
+// however reported health status should not come into effect.
+func IsElegibleForHealthStatusUpdates(wle *config.Config) bool {
```

Comment:
```suggestion
func IsEligibleForHealthStatusUpdates(wle *config.Config) bool {
```

nit - predates your fix.

Reply:
done